File: Leetcode.CPP/Leetcode1460.cpp
#include <vector>
using std::vector;
class Leetcode1460 {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        if (target.size() != arr.size()) return false;

        vector<int> dictT(1001, 0);
        vector<int> dictA(1001, 0);

        for (int i = 0; i < arr.size(); i++) {
            dictT[target[i]]++;
            dictA[arr[i]]++;
        }
        for (int i = 0; i < 1001; i++) {
            if (dictT[i] != dictA[i]) return false;
        }

        return true;
    }
};

File: C#/Leetcode1460.cs


namespace Leetcode.CSharp.Solutions; 
public class Leetcode1460 {
    public bool CanBeEqual(int[] target, int[] arr) {

        if(target.Length != arr.Length) return false;
        // len : 1 ~ 1000
        int[] dictT = new int[1000];
        int[] dictA = new int[1000];

        for (int i = 0; i < target.Length; i++) {
            dictT[target[i]]++;
            dictA[arr[i]]++;
        }
        for (int i = 0; i < 1000; i++) {
            if (dictA[i] != dictT[i]) return false;
        }
        return true;
    }
}
